test/units/executor/test_action_write_locks.py

```diff
@@ -0,0 +1,87 @@
+# (c) 2021, Kevin Carter <kecarter@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+
+from multiprocessing import Process
+from multiprocessing.managers import DictProxy
+
+import pytest
+
+from ansible.executor import action_write_locks
+from ansible.module_utils import six
+
+
+ACTION_LOCK = action_write_locks.action_lock
+ACTION_WRITE_LOCKS = action_write_locks.action_write_locks
+
+
+def _run_threads(test_int):
+    """Run thread test.
+
+    Test that that the original value is in our shared
+    dictionary and that we can add items into the cache
+    from within a process.
+    """
+    assert "testX" in ACTION_WRITE_LOCKS
+    ACTION_WRITE_LOCKS["test%i" % test_int] = ACTION_LOCK()
+
+
+def test_write_lock():
+    """Test that the write lock is created with default items."""
+    default_locks = {
+        "copy",
+        "file",
+        "setup",
+        "slurp",
+        "stat",
+    }
+    assert default_locks <= set(ACTION_WRITE_LOCKS)
+
+    if not six.PY2:
+        assert type(ACTION_WRITE_LOCKS) == DictProxy
+    else:
+        assert type(ACTION_WRITE_LOCKS) == dict
```

Comment:
I think I'd still prefer a single assertion
```suggestion
    expected_type = dict if six.PY2 else DictProxy
    assert type(ACTION_WRITE_LOCKS) == expected_type
```

Reply:
++ the gh review process wouldn't let me commit that one initial... 

I'm currently squashing the commits and will re-push shortly.